tools/pkg-install.cmake

```diff
@@ -0,0 +1,111 @@
+cmake_minimum_required(VERSION 3.12)
+
+# Default values for customizable paths.
+set(PACKAGE_LOCK_PATH "package.lock" CACHE FILEPATH "Path to the package.lock file.")
+set(PACKAGES_DIR ".packages" CACHE PATH "Directory to store cloned packages.")
+set(QUIET FALSE CACHE BOOL "Suppress output.")
+
+if (QUIET)
+  # Run this script again but with OUTPUT_QUIET set.
+  execute_process(
+    COMMAND "${CMAKE_COMMAND}"
+      "-DPACKAGE_LOCK_PATH=${PACKAGE_LOCK_PATH}"
+      "-DPACKAGES_DIR=${PACKAGES_DIR}"
+      "-DQUIET=FALSE"
```

Comment:
This looks a little bit odd.

Reply:
It's a recursive call to itself (thus setting the 'QUIET' to false to avoid an infinite loop.
Note that it has the "OUTPUT_QUIET" option set (line 16).
Added a comment.